src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs

```diff
@@ -178,6 +179,24 @@ public class SetAzureSqlManagedInstance : ManagedInstanceCmdletBase
         [Parameter(HelpMessage = "Skip confirmation message for performing the action")]
         public SwitchParameter Force { get; set; }
 
+        /// <summary>
+        /// Gets or sets the instance compute generation
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The compute generation for the instance.")]
+        [ValidateNotNullOrEmpty]
```

Comment:
Is it possible to add custom validators as attributes above fields?

Reply:
I've checked SQL DB, instance pools and managed instance new and set cmdlets and couldn't find the example of that so I decided to go this way. If it is possible, it is not common.

Reply:
AFAIK there is a completer for resource group name, if that exists then it must just be an ARM call so you could also call capabilities to find edition, sku name, compute gen, etc. So probably possible. But let's not block release on it :)